### PR TITLE
Init documentation

### PR DESCRIPTION
**Pull request by MisterVants**
Adds the first code documentation snippets to the client `init` and the `AppMetadata` object. Updates the readme to match the new signatures.

### CODE REVIEW

**Review by llbartekll — COMMENTED** [review]
@MisterVants in case ur still waiting for my review, just see the comments

File: Sources/WalletConnect/Types/Common/AppMetadata.swift [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
         self.name = name
         self.description = description
         self.url = url
         self.icons = icons
     }
-    
-    public let name: String?
-    public let description: String?
-    public let url: String?
-    public let icons: [String]?
```
[the remark this answers is not in the document]
Comment (MisterVants) [reply Sources/WalletConnect/Types/Common/AppMetadata.swift:15]: That's true, I'll check this out with Pedro

File: Sources/WalletConnect/WalletConnectClient.swift [hunk 2]
```diff
@@ -23,7 +23,30 @@ public final class WalletConnectClient {
...
+    /**
+     Initializes a WalletConnect client.
+     
+     - parameters:
+        - metadata: The App's information that is shared with connected peers.
+        - apiKey: The API key that identifies the client.
+        - isController: Whether this client should be responsible to control permissions on a session.
+        - relayURL: The URL of the relay server this client will connect to.
+     
+     You need to register your app and obtain a production API key to run the WalletConnect protocol. You can sign up for
```
[the remark this answers is not in the document]
Comment (MisterVants) [reply Sources/WalletConnect/WalletConnectClient.swift:36]: Good point, we need to change the doc as it was decided to make API key optional

**Comment by pedrouid** [comment]
Hey @MisterVants @llbartekll, the Swift SDK is different than the Javascript SDK because it cant read from the browser to fill the AppMetadata

So while you are correct that apiKey, relayUrl and controller can be optional. You still need to require the appMetadata to properly identify the app using the client

**Comment by llbartekll** [comment]
> So while you are correct that apiKey, relayUrl and controller can be optional. You still need to require the appMetadata to properly identify the app using the client
@pedrouid this is correct we will require appMetadata but can we assume that peer client will populate all the metadata properties(name, description, url, icons)? the case is that with this chance if one of them is nil than we cannot decode the whole object.

**Comment by pedrouid** [comment]
It depends on where you are referring to in the protocol. You should always follow the protocol to make assumptions about the peer's communication

For the pairing state you can have undefined (or empty object) for the metadata

For the session participant metadata you will always have defined metadata

**Comment by MisterVants** [comment]
There's just a small difference between an empty (null) metadata object and a metadata object with null fields. If there IS a metadata object, can we assume that all fields (name, desc, url, icons) will be populated / will not be null? (empty strings / arrays are ok, null values are the problem)

**Comment by MisterVants** [comment]
Closing for now, will come back later to get the code documentation and update it to fit the current API.

**Pull request closed without merging**